### PR TITLE
feat: 💡 mobile responsive collection overview

### PR DESCRIPTION
**Pull request by OyinloluB**
## Why?

Setting up mobile screen styles for collection overview

## How?

- Created a `useMediaQuery` hook for handling specific styling cases
- Conditionally displayed elements depending on screen size
- Modified styles for mobile screen sizes
- Hid filter section for mobile screens

## Tickets?

- [Notion](https://www.notion.so/Nice-to-have-Mobile-responsiveness-7e7a516084ec41d9989b596d8111e83d)

## Contribution checklist?

- [x] The commit messages are detailed
- [x] It does not break existing features (unless required)
- [x] I have performed a self-review of my own code
- [ ] Documentation has been updated to reflect the changes
- [ ] Tests have been added or updated to reflect the changes
- [x] All code formatting pass
- [x] All lints pass

## Demo?

Screenshot 2022-07-06 at 01 59 25

### CODE REVIEW

File: src/hooks/use-media-query.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { useState, useEffect } from 'react';
+
+const useMediaQuery = (query: string) => {
+  const [matches, setMatches] = useState(false);
```
Comment (prasanthLalapeta) [inline src/hooks/use-media-query.ts:4]: Shall we update state name to more developer friendly, at the moment I don't understand what is matches

File: src/components/icons/index.ts [hunk 2]
```diff
@@ -29,6 +29,9 @@ import { IoWarningOutline } from '@react-icons/all-files/io5/IoWarningOutline';
 import { FiUpload } from '@react-icons/all-files/fi/FiUpload';
 import { FiDownload } from '@react-icons/all-files/fi/FiDownload';
 import { AiFillPicture } from '@react-icons/all-files/ai/AiFillPicture';
+import { IoMenuSharp } from '@react-icons/all-files/io5/IoMenuSharp';
+import { IconWorld } from '@tabler/icons';
```
[the remark this answers is not in the document]
Comment (OyinloluB) [reply src/components/icons/index.ts:33]: Couldn't get that specific icon from the react-icons that we currently use, what do you suggest?

File: src/hooks/use-media-query.ts (hunk 1 above)
Comment (OyinloluB) [reply src/hooks/use-media-query.ts:4]:
```
    const media = window.matchMedia(query);
    if (media.matches !== matches) {
      setMatches(media.matches);
    }
```

It basically checks to see if there is a match, what name would you suggest?

File: src/components/icons/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (prasanthLalapeta) [reply src/components/icons/index.ts:33]: Let's use custom icon by importing icon in `src/components/icons/custom`

**Review by prasanthLalapeta — APPROVED** [review]

**Pull request merged**